### ISSUE TITLE
Collection "table" preset breaks when user lacks permission on related collection

### ISSUE THREAD

**Issue by colemahatmccreary** [directus/directus#12469]
### Preflight Checklist

- [x] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes my problem.

### Describe the Bug

If a collection's "Table" preset contains fields which reference another collection that the user lacks 'Read' access to, the Table returns every record as having all empty fields (except Primary Key).

<img width="1087" alt="Screen Shot 2022-03-29 at 1 41 40 PM" src="https://user-images.githubusercontent.com/40645344/160708975-d55a1427-7ef4-412a-a19a-a216e58dfa40.png">

I should note that this came up because the Collection has a default Table preset which is automatically applied to "All" users when they first join the app (irrespective of their role).

The remedy is probably something like: the fields requested by any table view should first filter out any fields/collections that the current role lacks access to.

I should ALSO note that this can technically be avoided by ensuring that the role's O2M field permissions match the permission granted on the related collection but this effectively forces you to make M(N + 1) permissions updates as relationships between the inaccessible collection and other (accessible) collections are created (with N being the number of relationships and M being the number of roles)

### To Reproduce

1. Create a 'user' collection with the following fields:
- first_name
- last_name
- orders (O2M - 'order')

2. Create an 'order' collection with the following fields:
- total
- user_id (M2O -> 'user')'

3. Set up display templates for the 'user' collection's 'orders' field (seems issue doesnt happen if no nested fields are included in the table's query) 

4. Create a 'Table' preset for the 'user' collection with all 3 fields visible - assign it to 'All' users.

5. Create a new role with read access to the 'user' collection and no access on the 'order' collection

6. Invite a new user and assign them the new role

7. View the 'user' collection's content (all non-PK fields should be blank because default preset requested a restricted O2M field)

8. Remove the 'orders' field from the collection's Table view (all fields should now be visible)

### Errors Shown

Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'related_collection')
    at fs (/admin/assets/index.262f690c.entry.js:43:209664)
    at Object.fields (/admin/assets/index.262f690c.entry.js:43:212938)
    at /admin/assets/index.262f690c.entry.js:43:167925
    at Proxy.map (<anonymous>)
    at sn (/admin/assets/index.262f690c.entry.js:43:167625)
    at S3.fn (/admin/assets/index.262f690c.entry.js:541:75152)
    at S3.run (/admin/assets/vendor.a27160c3.js:2:12019)
    at XF.get value [as value] (/admin/assets/vendor.a27160c3.js:2:20264)
    at wm (/admin/assets/vendor.a27160c3.js:2:19290)
    at x (/admin/assets/use-sync.33f9b39d.js:1:3935)

### What version of Directus are you using?

9.7.1 (Standard Cloud)

### What version of Node.js are you using?

Standard Cloud

### What database are you using?

Standard Cloud

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying Directus?

Standard Cloud

**Comment by jaads** [comment]
Can I give it a try as my first issue?

**Comment by licitdev** [comment]
@jaads Yes of course! 👍

**Comment by jaads** [comment]
I created a setup as described. In the current version, the rows are not rendered at all but a similar error is thrown and the app is crashing. I tried the same setup on a slightly older version and there I got the same bug how it was described above, so where all table rows got rendered but without the proper values in it, except the value for the primary key field. This proved that I had the correct setup. However, I concentrated on the current version obviously. 

I could get rid of the error by using optional chaining (`?.`) in the line which caused the error. With that, all rows get rendered correctly again and the app doesn't crash anymore. The values of the related field (to which no access was given) are now "`--`". Below is a demonstration of the fixed version.

https://user-images.githubusercontent.com/34009329/196892727-e2f977ec-372c-435b-b8e9-f055417aa26a.mp4

Although this fixed the bug, I was thinking about going a step further and get rid of the field in the preset to which the access is denied for the user. I can do so e.g. within the engines preset-service. At the same time I think this use case is more of an edge case, since a user would normally assign a preset to a role which fits the permissions of that role. Hence I would actually leave it as it is for now.
